Sources/MongoSwift/MongoClient.swift

```diff
@@ -14,6 +14,10 @@ public struct MongoClientOptions: CodingStrategyProvider {
     /// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/#urioption.compressors
     public var compressors: [Compressor]?
 
+    /// Specifies the time, in milliseconds, for an individual connection from your connection pool to establish a TCP
+    /// connection to the MongoDB server before timing out.
+    public var connectTimeoutMS: ChannelOptions.Types.ConnectTimeoutOption?
+
```

Comment:
In order for this option to be used by the driver, we'll need to pass it in to libmongoc. The way we do that is via the various `applyAndValidateXOptions` methods in `ConnectionString.swift`. For this option in particular, we'll want to add logic into `applyAndValidateConnectionPoolOptions`.

Once that's done, can we add a test to verify that the setting works? We'll probably want to put it into `SyncMongoClientTests.swift` (so that it will test both sync + async API).

An example test could be something like:
1. enable a [failCommand failpoint](https://github.com/mongodb/mongo/wiki/The-%22failCommand%22-fail-point) on the `isMaster` command with blockTimeMS = 500
    - This will ensure that a connection attempt takes at least 500ms, since the `isMaster` command is always run as part of the handshake for a new connection
2. create a client with `connectTimeoutMS` = 100
3. try to run a `ping` command and assert that a failure occurs
4. create a client with `connectTimeoutMS` = 1000
5. try to run a `ping` command and assert that it succeeds

For an example of a test that makes use of a failpoint, check out [`testCursorTerminatesOnError`](https://github.com/mongodb/mongo-swift-driver/blob/main/Tests/MongoSwiftSyncTests/MongoCursorTests.swift#L243).

In order to set failpoints, be sure to start your `mongod` locally with the `--setParameter enableTestCommands=1` option.